code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -631,11 +631,20 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 	var/close_sound = list('sound/items/zippo_close.ogg')
 	fuel = 100 //Zippos da bes
 
+/obj/item/weapon/lighter/red
+	icon_state = "lighter-r"
+/obj/item/weapon/lighter/cyan
+	icon_state = "lighter-c"
+/obj/item/weapon/lighter/yellow
+	icon_state = "lighter-y"
+/obj/item/weapon/lighter/green
+	icon_state = "lighter-g"
+
 /obj/item/weapon/lighter/random/New()
-	. = ..()
-	var/color = pick("r","c","y","g")
-	icon_state = "lighter-[color]"
-	update_brightness()
+	var/color = pick("red","cyan","yellow","green")
```

Comment:
Self deleting objects like this are terrible because a contents list isn't the only thing that can create on-creation references.

Make it set icon state procedurally and keep the subtypes, but don't do this.